### PR TITLE
Remove resolve_2many_commands

### PR DESCRIPTION
**Pull request by Elkasitu**
Linked to https://github.com/odoo/enterprise/pull/9409/

### CODE REVIEW

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -3343,7 +3342,7 @@ def field_will_change(line, field_name):
                 return line[field_name].id != vals[field_name]
             if field.type in ('one2many', 'many2many'):
                 current_ids = set(line[field_name].ids)
-                after_write_ids = set(r['id'] for r in line.resolve_2many_commands(field_name, vals[field_name], fields=['id']))
+                after_write_ids = set(getattr(line.new({field_name: vals[field_name]}), field_name).ids)
```
Comment (rco-odoo) [inline addons/account/models/account_move.py]: Avoid using `getattr` to get a field by its name, use the `[]` operator instead:
```suggestion
                after_write_ids = set(line.new({field_name: vals[field_name]})[field_name].ids)
```

File: addons/account/models/account_move.py [hunk 2]
```diff
@@ -3319,9 +3319,8 @@ def create(self, vals_list):
                 tax = repartition_line.invoice_tax_id or repartition_line.refund_tax_id
                 vals['tax_exigible'] = tax.tax_exigibility == 'on_invoice'
             elif vals.get('tax_ids'):
-                tax_ids = [v['id'] for v in self.resolve_2many_commands('tax_ids', vals['tax_ids'], fields=['id'])]
-                taxes = self.env['account.tax'].browse(tax_ids).flatten_taxes_hierarchy()
-                vals['tax_exigible'] = not any(tax.tax_exigibility == 'on_payment' for tax in taxes)
+                taxes = self.new({'tax_ids': vals['tax_ids']}).tax_ids.flatten_taxes_hierarchy()
+                vals['tax_exigible'] = len(taxes.filtered(lambda t: t.tax_exigibility == 'on_payment')) < 1
```
Comment (rco-odoo) [inline addons/account/models/account_move.py]: Why did you change this?  It was working fine, IMO.
```suggestion
                vals['tax_exigible'] = not any(tax.tax_exigibility == 'on_payment' for tax in taxes)
```

File: addons/payment/models/payment_acquirer.py [hunk 3]
```diff
@@ -865,8 +865,8 @@ def _cron_post_process_after_done(self):
     @api.model
     def _compute_reference_prefix(self, values):
         if values and values.get('invoice_ids'):
-            many_list = self.resolve_2many_commands('invoice_ids', values['invoice_ids'], fields=['name'])
-            return ','.join(dic['name'] for dic in many_list)
+            invoices = self.new({'invoice_ids': values['invoice_ids']})
+            return ','.join(invoices.invoice_ids.mapped('name'))
```
Comment (rco-odoo) [inline addons/payment/models/payment_acquirer.py]: The variable `invoices` is misleading in this context
```suggestion
            invoices = self.new({'invoice_ids': values['invoice_ids']}).invoice_ids
            return ','.join(invoices.mapped('name'))
```

File: addons/sale/models/payment.py [hunk 4]
```diff
@@ -138,8 +138,8 @@ def _invoice_sale_orders(self):
     def _compute_reference_prefix(self, values):
         prefix = super(PaymentTransaction, self)._compute_reference_prefix(values)
         if not prefix and values and values.get('sale_order_ids'):
-            many_list = self.resolve_2many_commands('sale_order_ids', values['sale_order_ids'], fields=['name'])
-            return ','.join(dic['name'] for dic in many_list)
+            sale_orders = self.new({'sale_order_ids': values['sale_order_ids']})
+            return ','.join(sale_orders.sale_order_ids.mapped('name'))
```
Comment (rco-odoo) [inline addons/sale/models/payment.py]: The variable `sale_orders` is misleading, because `self` is not the model of sales orders.

File: addons/stock/models/stock_warehouse.py [hunk 5]
```diff
@@ -146,8 +146,8 @@ def write(self, vals):
             warehouses._update_reception_delivery_resupply(vals.get('reception_steps'), vals.get('delivery_steps'))
 
         if vals.get('resupply_wh_ids') and not vals.get('resupply_route_ids'):
-            resupply_whs = self.resolve_2many_commands('resupply_wh_ids', vals['resupply_wh_ids'])
-            new_resupply_whs = self.browse([wh['id'] for wh in resupply_whs])
+            resupply_whs = self.new({'resupply_wh_ids': vals['resupply_wh_ids']})
+            new_resupply_whs = self.browse(resupply_whs.resupply_wh_ids.ids)
```
Comment (rco-odoo) [inline addons/stock/models/stock_warehouse.py]: Same as above.

File: addons/account/models/account_move.py (hunk 2 above)
Comment (Elkasitu) [reply addons/account/models/account_move.py]: I think this was the product of a bad merge conflict resolution
